ui/lib/pki/addon/routes/roles/index.js

```diff
@@ -19,5 +23,9 @@ export default class RolesIndexRoute extends Route {
           throw err;
         }
       });
+    return RSVP.hash({
```

Comment:
Jordan, you might have some comments on this. I tried not using hash, but had issues with it returning a promise and not a class. I tried a then, and that didn't work.

Reply:
Since `modelFor` does not return a promise you could return an object in the `then` of query:
```js
return this.store
  .query('pki/pki-role-engine', { backend: this.secretMountPath.currentPath })
  .then((model) => {
    return { model, parentModel: this.modelFor('roles') };
  })
  .catch((err) => {
    if (err.httpStatus === 404) {
      return [];
    } else {
      throw err;
    }
  });
```
I don't mind using `RSVP.hash` though too. It just might not be immediately clear to someone who is unfamiliar with the method.

I would avoid using `let` to assign those variables since they are not being reassigned if you decide to leave as is.

Reply:
Awesome, thank you. I see now how to structure the then. I prefer this method only because it doesn't rely on an import, it's pure js.